### PR TITLE
[ALL] Add support for simplified local documentation build using GNU make.

### PR DESCRIPTION
**Pull request by claudiocebpaz**
Signed-off-by: Claudio Ceballos Paz <claudioceballospaz@MacBook-Air-de-Claudio.local>

Adds support for the local build using GNU make.

It works by calling from the root of the repo:
make docs lang=YOUR_LANG_DIR

example:
```
make docs lang=es
make clean lang=es
```

### CODE REVIEW

**Comment by davidkhala** [comment]
extends from #83

**Comment by lindluni** [comment]
I'm obviously not the stakeholder here, but what about something like this for doc writers by just creating appropriate targets for each language, in this case:

```
docs-lang-es: LANG = es
docs-linkcheck-lang-es: LANG = es

docs-lang-%:
    ifndef LANG
        $(error "Please use a language specific `make` target such as: `docs-lang-<language>` where `<language> is the `i18n` short code for your language)
    endif
    @echo "Generating translation for $(LANG)"
    @docker run -v $$(pwd)/docs:/docs hyperledger-fabric.jfrog.io/fabric-tox sh -c 'cd /docs/locale/$(LANG)/ && tox -e docs'

docs-linkcheck-lang-%:
	@docker run -v $$(pwd)/docs:/docs hyperledger-fabric.jfrog.io/fabric-tox sh -c 'cd /docs/locale/$(LANG)/ && tox -e docs-linkcheck'
```

Then future implementations would only need to implement the `docs-lang-<language>` and `docs-checklink-lang-<language>`, and the doc translators would no longer need to pass parameters into the make targets nor would they need to be aware of this. The wilcard targets would handle the rest.

**Comment by lindluni** [comment]
I'm not trying to step on anyone's toes here, I can just say from experience from Fabric that Doc maintenance gets out of hand very quickly, and providing the simplest possible path for non-technical resources is something we are working to do better in the fabric repo (especially around doc, as it's gotten nearly unmaintainable).

**Comment by davidkhala** [comment]
> I'm not trying to step on anyone's toes here, I can just say from experience from Fabric that Doc maintenance gets out of hand very quickly, and providing the simplest possible path for non-technical resources is something we are working to do better in the fabric repo (especially around doc, as it's gotten nearly unmaintainable).

What is the major maintain issue in fabric docs in core repos? I guess it is more on content conflict than code structure, is it?

**Comment by claudiocebpaz** [comment]
@btl5037 Thanks for your review. I'll take that into account and send this again in that format.
I was trying to avoid changing the makefile every time a new translation is added. 
But I think that consistency is more important. 

I'll change the makefile using your feedback.

**Comment by lindluni** [comment]
> @btl5037 Thanks for your review. I'll take that into account and send this again in that format.
> I was trying to avoid changing the makefile every time a new translation is added.
> But I think that consistency is more important.
> 
> I'll change the makefile using your feedback.

You could also avoid the need to add new targets altogether by simply keeping the wildcard targets and parsing the wildcard out of the target and checking to see if the directory exists.

The upside to having individual targets though is typing `docs-lang-` and pressing `tab` will give you all the available languages, which is something you wouldn't get with using a parameter passed to the make target or using purely a wildcard target

**Comment by claudiocebpaz** [comment]
Thanks, @btl5037 I think the upside reason is good enough to have it that way.

**Comment by lindluni** [comment]
> > I'm not trying to step on anyone's toes here, I can just say from experience from Fabric that Doc maintenance gets out of hand very quickly, and providing the simplest possible path for non-technical resources is something we are working to do better in the fabric repo (especially around doc, as it's gotten nearly unmaintainable).
> 
> What is the major maintain issue in fabric docs in core repos? I guess it is more on content conflict than code structure, is it?

The biggest issue we had (and we've corrected it in many places) is that expecting the user to pass in a variable is unintuitive, as there is no place that it is naturally codified, so you need to know the `i18n` code ahead of time. Whereas adding them as targets means simple make commands (or looking at the makefile) present you with all of your available options with no expectation that you have prior knowledge of how the process works. We use to have a lot of targets like this, in particular around `ldflags` and also with `race` flags on test targets. People were getting unexpected results (often discovered too late) because these parameters weren't codified.

We can also be bit by a growing list of parameters. While one parameters might seem manageable, if our process changes and you now have to specify two or three parameters it quickly becomes a nightmare to manage, simply attaching the expected parameters to a simple make target takes all the complexity away.

For people with deeply technical backgrounds like you and I, it seems very trivial, but @pamandrejko can attest to the fact that no matter how simple we make the process, contributors will invariably turn it into "quantum mechanics". Pam spends a ton of time explaining simple things to doc contributors, such as signing-off your commits, or the fact you need to rebuild the metrics doc if you update metrics (because this is one area we don't naturally check or codify, and thus its a black box to contributors).

As the surface area for internationalization continues to grow, it seems natural to make the process as simple as possible for contributors, even if it means we the maintainers have to do a little extra work in codifying some of this complexity. If we get to dozens of languages, issues being opened to ask why the expected language wasn't built will flourish, or worse yet, people will push valid changes and then simply abandon them in PR (this happens to us dozens of times a month) because they didn't build them locally because it wasn't ridiculously simple.

Sorry for the text wall David, needed to hit my monthly quota 😃

**Comment by claudiocebpaz** [comment]
New makefile structure is pushed. Please review and advise.

now works like this:

```
make docs-lang-es
docs-linkcheck-lang-es
docs-clean-lang-es
```

Thanks.

File: Makefile
Comment (davidkhala) [inline Makefile]: Sorry that I am not expert on Make. What will be the no_value content?

**Comment by davidkhala** [comment]
@claudiocebpaz could you also provide a change to AZP configuration file about incorporating it in build cycle?

File: Makefile
Comment (lindluni) [reply Makefile]: I think its being used as a zero-value. Make will parse the file first (thus initially setting `TRANSLATION = no_value`) and then the target is called. If it's a valid target its `TRANSLATION` is overwritten with an actual value.

But is there a reason not to just use `ifndef` like you were before?

Comment (claudiocebpaz) [reply Makefile]: Hello, @davidkhala and @btl5037 .

As Bret said, I'm using this assignment as a zero value to verify if the translation lang exists.

What happened here is that I tried Bret's suggestion of using "ifndef" but I couldn't make it work as proposed. 

A few things:
First, What Bret suggested was great, but using the LANG variable wasn't possible because Make would interpret the LANG environment variable from the console.
So, if no LANG was passed, it took the default console language.

Second, When I used ifndef, and the variable wasn't set in the very beginning, I got an error altogether. 

This didn't work:
```
docs-lang-es: TRANS = es

docs-lang-%:
    ifndef TRANS
        $(error "Please use a language specific `make` target such as: `docs-lang-<language>` where `<language> is the `i18n` short code for your language)
    endif
```

This worked:
```
TRANS = us
docs-lang-es: TRANS = es

docs-lang-%:
    ifndef TRANS
        $(error "Please use a language specific `make` target such as: `docs-lang-<language>` where `<language> is the `i18n` short code for your language)
    endif
```

After research, I found out that all ifndef are evaluated in a first pass of the makefile.

check this, please:
https://stackoverflow.com/questions/11994029/makefile-ifeq-when-are-they-evaluated

Therefore, the workaround that I found was to do this target with the sh script in it.

Any suggestions?

**Comment by claudiocebpaz** [comment]
@davidkhala 
I PR this: https://github.com/hyperledger/fabric-docs-i18n/pull/90 for AZP

Is there something I'm missing?

**Comment by davidkhala** [comment]
> @davidkhala
> I PR this: #90 for AZP
> 
> Is there something I'm missing?

LGTM，but as long as the mechanism changed my Approve and Merge rights is now removed. 
You have to ask review from @hyperledger/fabric-core-doc-maintainers 
And one more, if you are ready to be reviewed, please squash your commit and force push again.

**Review by davidkhala — APPROVED** [review]
It has been for a while. @btl5037 Bret, I would like to merge it first to keep community healthy. And please help to raise another PR if any further optimize you suggest.

**Comment by lindluni** [comment]
Merging, however we do need to add `tox.ini` files to the rest of the projects

**Review by lindluni — APPROVED** [review]

**Comment by davidkhala** [comment]
> Merging, however we do need to add `tox.ini` files to the rest of the projects

Dear Bret, would you mind creating an issue for it? It will be better if you could specific where should we added to. 
Edit: it is found done by #137

**Pull request merged**
